### PR TITLE
[office-js][office-js-preview] (Outlook) Deprecate entity-based contextual APIs

### PR DESCRIPTION
**Pull request by samantharamon**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: None.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Review by Rick-Kirkham — APPROVED** [review]

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -18433,8 +18358,8 @@ declare namespace Office {
          *
          * @returns
          * An object that contains arrays of strings that match the regular expressions defined in the manifest XML file.
-         * The name of each array is equal to the corresponding value of the RegExName attribute of the matching `ItemHasRegularExpressionMatch` rule
-         * or the `FilterName` attribute of the matching `ItemHasKnownEntity` rule. For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur in the property
+         * The name of each array is equal to the corresponding value of the RegExName attribute of the matching `ItemHasRegularExpressionMatch` rule.
+         * For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur in the property
          * of the item that is specified by that rule. The `PropertyName` simple type defines the supported properties.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:18363]: here and other instances

```suggestion
         * of the item that's specified by that rule. The `PropertyName` simple type defines the supported properties.
```

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -18540,9 +18447,9 @@ declare namespace Office {
          *
          * @returns
          * An object that contains arrays of strings that match the regular expressions defined in the manifest XML file.
-         * The name of each array is equal to the corresponding value of the `RegExName` attribute of the matching `ItemHasRegularExpressionMatch` rule or
-         * the `FilterName` attribute of the matching `ItemHasKnownEntity` rule. For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur
-         * in the property of the item that is specified by that rule. The `PropertyName` simple type defines the supported properties.
+         * The name of each array is equal to the corresponding value of the `RegExName` attribute of the matching `ItemHasRegularExpressionMatch` rule.
+         * For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur in the property of the item that is specified by that rule.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:18451]:
```suggestion
         * For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur in the property of the item that's specified by that rule.
```

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -19124,46 +19029,52 @@ declare namespace Office {
      * An array of `PhoneNumber` objects containing the phone numbers found in an email message is returned in the `phoneNumbers` property of the
      * `Entities` object that is returned when you call the `getEntities` method on the selected item.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:19030]:
```suggestion
     * `Entities` object that's returned when you call the `getEntities` method on the selected item.
```

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -21143,6 +21054,13 @@ declare namespace Office {
      * The list of tasks suggested in an email message is returned in the `taskSuggestions` property of the {@link Office.Entities | Entities} object
      * that is returned when the `getEntities` or `getEntitiesByType` method is called on the active item.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:21055]:
```suggestion
     * that's returned when the `getEntities` or `getEntitiesByType` method is called on the active item.
```

File: types/office-js/index.d.ts [hunk 5]
```diff
@@ -12317,9 +12223,9 @@ declare namespace Office {
          *
          * @returns
          * An object that contains arrays of strings that match the regular expressions defined in the manifest XML file.
-         * The name of each array is equal to the corresponding value of the `RegExName` attribute of the matching `ItemHasRegularExpressionMatch` rule or
-         * the `FilterName` attribute of the matching `ItemHasKnownEntity` rule. For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur
-         * in the property of the item that is specified by that rule. The `PropertyName` simple type defines the supported properties.
+         * The name of each array is equal to the corresponding value of the `RegExName` attribute of the matching `ItemHasRegularExpressionMatch` rule.
+         * For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur in the property of the item that is specified by that rule.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:12227]:
```suggestion
         * For an `ItemHasRegularExpressionMatch` rule, a matching string has to occur in the property of the item that's specified by that rule.
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]
In case you opt to make this change - a number of places say "that is" that can be updated to "that's". I didn't add suggestions to all of them

**Pull request merged**